### ISSUE TITLE
Cleanup call to canvas.Refresh()

### ISSUE THREAD

**Issue by Bluebugs** [fyne-io/fyne#3823]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

Call to canvas.Refresh should be an exception and should be called only by the widget on itself when the content change and require a rendering update. Our code base evolved and we need to review all use place.

### Is it possible to construct a solution with the existing API?

Totally, just need to clean up.

### Describe the solution you'd like to see.

The idea would be to rename canvas.Refresh to something that imply that you really need to call Refresh and deprecate canvas.Refresh.

**Comment by Jacalz** [comment]
On the topic of this, should container refresh all its objects and then refresh the canvas afterwards?
https://github.com/fyne-io/fyne/blob/018a2e14545aa1e854bea3b1d590be272ed62080/container.go#L118-L123

**Comment by andydotxyz** [comment]
I don't think it should have to trigger canvas - as the container has no visual representation...
